### DISCUSSION TITLE
Confused about async/sync dmenu mode

### DISCUSSION THREAD

**Discussion by kraftwerk28** [davatorium/rofi#1567] [Q&A] [answered]
As far as I understand from `rofi(1)`, by default `rofi -dmenu` doesn't wait until stdin EOF occurs, unless `-sync` flat provided.
Therefore,
```bash
(echo first; sleep 1; echo second) | rofi -dmenu
```
should launch rofi immediately, with one entry, `first`. Then after 1 second, `second` should be added to the list, right? But for me, it still waits 1 second and then launches rofi w/ two entries.

**Comment by DaveDavenport** [comment, marked as answer]
See the following setting:
```
	-async-pre-read [number]               Read several entries blocking before switching to async mode

```

**Reply by kraftwerk28** [reply]
That works, thank you!
